app/src/main/java/org/notests/rxfeedbackexample/github/paginated/search/GithubPaginatedSearch.kt

```diff
@@ -90,17 +90,17 @@ fun State.Companion.reduce(state: State, event: Event): State =
                         state.copy(results = state.results.plus(event.response.value.first),
                                 shouldLoadNextPage = false,
                                 nextPageUrl = event.response.value.second,
-                                lastError = Optional.None())
+                                lastError = null)
                     is Result.Failure ->
                         state.copy(shouldLoadNextPage = false,
-                                lastError = Optional.Some(event.response.error))
+                                lastError = event.response.error)
                 }
         }
 
 // queries
-val State.loadNextPage: Optional<String>
+val State.loadNextPage: String?
```

Comment:
Now, on line 154 we pass null inside a chain because of this change:
```
it.map { it.loadNextPage }.drive { showQuery(it) }
```
We had optional here because in rxJava you'll get NPE immediately if you pass null through the chain. 
https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#nulls

This works here because we're using `Driver` that is has `try-catch` block inside it's `map` function so it swallows NPE but nulls are not supposed to be used in rxjava2.

That was also the reason why we used `Optional` for `nextPageUrl` and `lastError` (ok we currently don't map a state to these fields so that could remain nullable)

Reply:
@eliekarouz — @JurajBegovac is correct. Although you refactored feedback loops in a way that they don't require `Optional`s, this guy still needs it. 

Instead of removing an optional here, you should convert it on line 162.

Reply:
@JurajBegovac , @markotron 
Yeah you are right. I missed it in the map operator. Actually I didn't test the example yet.

Reply:
@JurajBegovac , @markotron , We can transform them to empty strings so we don't have to add the back the `Optional`. I know it's a bit dirty... A cleaner solution would be to add back the `Optional` but make it internal to the library or even public if you want.